Refactoring(convention included)

onikss793 opened the pull request:
1. changed service, controller, util convention when import
2. collection service refactoring usin Promis.all and map method

onikss793 committed:
while rebasing deleted category, collection, curriculum... query

onikss793 committed:
Crawler to crawler:

onikss793 committed:
# This is a combination of 2 commits.
# This is the 1st commit message:

Refactoring(convention included)

1. changed service, controller, util convention when import
2. collection service refactoring usin Promis.all and map method

# This is the commit message #2:

review applied except 'await' to services/collection

onikss793 committed:
conflict resolved

joonseokhu commented on services/collection.js:
```diff
@@ -12,27 +12,22 @@ exports.paginatedCollection = async (page, limit) => {
     .sort("-score");
 
-  let response = [];
+  return Promise.all(
```
Promise.all 함수는 배열로 묶인 프로미스들을 해결하는 함수이지만, 자기 자신도 마찬가지로 프로미스를 리턴합니다. await 키워드를 붙여야 돼요

onikss793 replied:
controllers/collection에서 저 paginatedCollection함수를 await로 받는 것과 관계 없이 저 함수 내에서도  await 를 해야 하는 것인가요?

vraimentres replied:
@onikss793 네 그렇습니다~

vraimentres replied:
[async/await 함수간 결과 사용 예시](https://gist.github.com/vraimentres/3d537d6ffe2dc938390de98e866d790b)
간단한 예시를 써보았는데 참고해보시고 도움이 되시면 좋겠습니다

joonseokhu replied:
아하 컨트롤러에서 저 promise를 그대로 받아서 await 로 풀고 계시군요! 괜찮을것같습니다~

joonseokhu commented on services/collection.js:
```diff
@@ -12,27 +12,22 @@ exports.paginatedCollection = async (page, limit) => {
+        };
+      } catch (err) {
+        console.log(err);
+      }
```
비동기 로직에서 발생한 에러는, 콘솔을 띄우고 무시되도록 처리하는 것 보다 바깥으로 에러를 전달해주는게 에러 관리에 더 좋을 수 있습니다

onikss793 committed:
Refactoring(convention included)

1. changed service, controller, util convention when import
2. collection service refactoring usin Promis.all and map method

saegeullee committed:
Implemented user authorization functionality

Implementations
1. Implemented user authorization handler loginRequired
2. Updated user signup error conditions
